libraries/mempool/mempool.cpp

```diff
@@ -22,435 +20,535 @@ namespace koinos::mempool {
 
 namespace detail {
 
-using namespace boost;
 using int128_t = boost::multiprecision::int128_t;
-using namespace std::chrono_literals;
 
-struct pending_transaction_object
+class mempool_impl final
 {
-   protocol::transaction                 transaction;
-   std::chrono::system_clock::time_point time;
-   nonce_type                            nonce;
-   uint64_t                              disk_storage_used;
-   uint64_t                              network_bandwidth_used;
-   uint64_t                              compute_bandwidth_used;
-
-   const transaction_id_type& id() const
-   {
-      return transaction.id();
-   }
+private:
+   state_db::database _db;
 
-   const account_type& payer() const
-   {
-      return transaction.header().payer();
-   }
+   crypto::multihash tmp_id( crypto::multihash id ) const;
+   state_db::state_node_ptr relevant_node( std::optional< crypto::multihash > id, state_db::shared_lock_ptr lock ) const;
+   state_db::state_node_ptr relevant_node( std::optional< crypto::multihash > id, state_db::unique_lock_ptr lock ) const;
+
+   void cleanup_account_resources( state_db::state_node_ptr node, const pending_transaction_record& pending_trx );
+
+   uint64_t remove_pending_transactions( state_db::state_node_ptr node, const std::vector< transaction_id_type >& ids );
+
+   bool check_pending_account_resources(
+      state_db::anonymous_state_node_ptr node,
```

Comment:
I don't think there is a need here for this specifically to be an `anonymouse_state_node_ptr`. An `abstract_state_node_ptr` would be sufficient and then would not require creating an anonymous state node for reading.